lib/details/view/details_page.dart

```diff
@@ -0,0 +1,142 @@
+import 'package:decimal/decimal.dart';
+import 'package:decimal/intl.dart';
+import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+import '../../shared/utils/utils.dart';
+import '../utils/details_arguments.dart';
+import '../utils/providers/providers.dart';
+import '../widgets/details_convert_button.dart';
+import '../widgets/details_info_row.dart';
+import '../widgets/line_chart.dart';
+import '../widgets/variation_info.dart';
+
+class DetailsPage extends HookConsumerWidget {
+  static const routeName = "/details";
+
+  const DetailsPage({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final arguments =
+        ModalRoute.of(context)!.settings.arguments as DetailsArguments;
```

Comment:
MaterialApp(
  // Provide a function to handle named routes.
  // Use this function to identify the named
  // route being pushed, and create the correct
  // Screen.
  onGenerateRoute: (settings) {
    // If you push the PassArguments route
    if (settings.name == PassArgumentsScreen.routeName) {
      // Cast the arguments to the correct
      // type: ScreenArguments.
      final args = settings.arguments as ScreenArguments;

      // Then, extract the required data from
      // the arguments and pass the data to the
      // correct screen.
      return MaterialPageRoute(
        builder: (context) {
          return PassArgumentsScreen(
            title: args.title,
            message: args.message,
          );
        },
      );
    }
    // The code only supports
    // PassArgumentsScreen.routeName right now.
    // Other values need to be implemented if we
    // add them. The assertion here will help remind
    // us of that higher up in the call stack, since
    // this assertion would otherwise fire somewhere
    // in the framework.
    assert(false, 'Need to implement ${settings.name}');
    return null;
  },
)
usa algo parecido com isso